src/Microsoft.AspNet.Mvc.Core/ModelBinding/MutableObjectModelBinder.cs

```diff
@@ -394,14 +383,17 @@ internal static PropertyValidationInfo GetPropertyValidationInfo(ModelBindingCon
                     propertyName);
 
                 // Execute validator (if any) to get custom error message.
-                IModelValidator validator;
-                if (validationInfo.RequiredValidators.TryGetValue(missingRequiredProperty, out validator))
+                var validatorProviderContext = new ModelValidatorProviderContext(propertyExplorer.Metadata);
+                bindingContext.OperationBindingContext.ValidatorProvider.GetValidators(validatorProviderContext);
+                var validator = validatorProviderContext.Validators.FirstOrDefault();
+
+                if (validator != null)
```

Comment:
Why run the first validator if it does not have `IsRequired == true`?

Further, the error (if any) will likely be incorrect and the "fall back" message better. Until and unless we add a new always-first validator for `[BindRequired]` cases, suggest removing lines 386 through 393 (new line numbers) entirely.